Add info about using native Docker on win10 and 2016

mstanleyjones opened the pull request:
### Describe the proposed changes

Clear up confusion about running Linux containers on Win10 and 2016

### Project version

n/a

### Related issue

n/a

### Related issue or PR in another project

https://github.com/docker/docker/issues/27668

### Please take a look

@thaJeztah @friism @londoncalling @happysysadm

friism commented on docker-for-windows/index.md:
```diff
@@ -25,6 +25,16 @@ Please read through these topics on how to get started. To **give us your feedba
 
+## Using pre-installed Docker on Windows Server 2016 or Windows 10
+
+Windows Server 2016 and Windows 10 include a native Docker daemon, which allows
```
@mstanleyjones I don't thinkg "include" is the best work - they still have to be installed. Maybe "Windows Server 2016 and Windows 10 can run the Docker daemon natively"

mstanleyjones replied:
Did some rewording / reorganization to try to make it clearer. Thanks.

londoncalling commented:
@mstanleyjones okay, I'm almost done with this, but let's work through the conflicts together tomorrow.

mstanleyjones commented:
@londoncalling I fixed the conflicts:

1. Checked out the branch.
2. Fetched my `origin`, rebased the branch on my `origin` (to get your commit)
3. Fetched my `upstream`, rebased the branch on my `upstream` (to get the conflict)
4. Resolved the conflict in `docker-for-mac/index.md` and `docker-for-windows/index.md`) by:
    - choosing your side of the conflict
    - Fixing the formatting of the note, using indentation to make sure the list items
       were inside the note sidebar
    - Removed the unnecessary `<br>` tags
    - Wrapping the lines at 80 characters, with hanging indent, so that it's easier to
       read the source and so that in the future it's easier to maintain
    - Changing the `*` bullets to `-` ones for personal preference

mstanleyjones commented on docker-for-mac/index.md:
```diff
@@ -54,18 +54,22 @@ For more about stable and beta channels, see the [FAQs](faqs.md#stable-and-beta-
-
 * **README FIRST for Docker Toolbox and Docker Machine users**: If you are already running Docker on your machine, first read [Docker for Mac vs. Docker Toolbox](docker-toolbox.md) to understand the impact of this installation on your existing setup, how to set your environment for Docker for Mac, and how the two products can coexist.
-
+<p />
```
I don't think these are needed.

mstanleyjones commented on docker-for-windows/index.md:
```diff
@@ -52,27 +51,46 @@ If you have not already done so, please install Docker for Windows. You can down
 
 * **README FIRST for Docker Toolbox and Docker Machine users**: Docker for Windows requires Microsoft Hyper-V to run. After Hyper-V is enabled, VirtualBox will no longer work, but any VirtualBox VM images will remain. VirtualBox VMs created with `docker-machine` (including the `default` one typically created during Toolbox install) will no longer start. These VMs cannot be used side-by-side with  ...
-
+<p />
```
I don't think these are needed either.

mstanleyjones commented on docker-for-windows/index.md:
```diff
@@ -52,27 +51,46 @@ If you have not already done so, please install Docker for Windows. You can down
 * Virtualization must be enabled. Typically, virtualization is enabled by default. (Note that this is different from having Hyper-V enabled.) For more detail see [Virtualization must be enabled](troubleshoot.md#virtualization-must-be-enabled) in Troubleshooting.
-
-* **What the install includes**: The installation provides [Docker Engine](https://docs.docker.com/engine/userguide/intro/), Docker CLI client, [Docker Compose](https://docs.docker.com/compose/overview/), and [Docker Machine](https://docs.docker.com/machine/overview/).
+<br><br>
```
I don't think these are needed either.

londoncalling commented:
LGTM
